pennylane/transforms/optimization/optimization_utils.py

```diff
@@ -79,6 +121,41 @@ def _fuse(angles_1, angles_2):
     return stack([z1, y, z2])
 
 
+def _fuse(angles_1, angles_2):
+    """Perform fusion of two angle sets. Separated out so we can do JIT with conditionals."""
+    # Compute the product of the quaternions
+    q = _quaternion_product(_zyz_to_quat(angles_1), _zyz_to_quat(angles_2))
+
+    # Convert the product back into the angles fed to Rot
+    y_arg = 1 - 2 * (q[1] ** 2 + q[2] ** 2)
```

Comment:
Similar thoughts re. code duplication: could we have these lines be moved somewhere else such that they are not duplicated across this function and `_fuse_jax`?